ui/src/main/webapp/src/app/components/analysis-context-form.component.ts

```diff
@@ -19,49 +19,57 @@ import {RouteFlattenerService} from "../services/route-flattener.service";
 import {WindupService} from "../services/windup.service";
 import {NotificationService} from "../services/notification.service";
 import {utils} from "../utils";
+import {RegisteredApplicationService} from "../services/registered-application.service";
+import {RegisteredApplication} from "windup-services";
 
 @Component({
     templateUrl: './analysis-context-form.component.html'
 })
-export class AnalysisContextFormComponent extends FormComponent implements OnInit, IsDirty
+export class AnalysisContextFormComponent extends FormComponent
+    implements OnInit, IsDirty, OnChanges
 {
     @ViewChild(NgForm)
-    private analysisContextForm:NgForm;
+    private analysisContextForm: NgForm;
 
     private _dirty: boolean = null;
 
     applicationGroup: ApplicationGroup = null;
 
-    analysisContext:AnalysisContext;
-    packageTree: Package[] = [];
+    analysisContext: AnalysisContext;
+
+    availableApps: RegisteredApplication[]
+        //= [<RegisteredApplication>{id: 1, title: "MyApp"}];
+    includedApps: RegisteredApplication[];
+
 
     /**
      * These two variables exist because we need for the item in the array not to just be a literal.
-     *
-     * This works around issues with JavaScript not being able to iterate over and modify a simple array of literals within
-     * a form easily.
-     *
+     * Workaround for JavaScript issues not able to iterate and modify a simple array of literals within a form easily.
      * See also: http://stackoverflow.com/questions/33346677/angular2-ngmodel-against-ngfor-variables
      */
     includePackages: Package[];
     excludePackages: Package[];
+    packageTree: Package[] = [];
 
-    configurationOptions:ConfigurationOption[] = [];
+    configurationOptions: ConfigurationOption[] = [];
 
-    private _migrationPathsObservable:Observable<MigrationPath[]>;
+    private _migrationPathsObservable: Observable<MigrationPath[]>;
```

Comment:
@klinki - I would love to have a code formatter that we could all consistently use. We all use different IDEs, though, I think, so that might be a little harder. It would probably have to be an external tool.